packages/react-components/react-datepicker/src/components/CalendarDay/CalendarDay.types.ts

```diff
@@ -25,18 +24,13 @@ export interface ICalendarDay {
 /**
  * {@docCategory Calendar}
  */
-export interface CalendarDayProps extends IBaseProps<ICalendarDay>, CalendarDayGridProps {
+export interface CalendarDayProps extends CalendarDayGridProps {
   /**
    * Optional callback to access the ICalendarDay interface. Use this instead of ref for accessing
    * the public methods and properties of the component.
    */
   componentRef?: React.RefObject<ICalendarDay>;
 
```

Comment:
It is safe to remove `IBaseProps` since it resolves to the same as the `componentRef` below and we are also overriding what `IBaseProps` adds by providing a specific `componentRef`.